src/bindings/fluid_ladspa.c

```diff
@@ -240,15 +231,18 @@ fluid_ladspa_fx_t *new_fluid_ladspa_fx(fluid_real_t sample_rate, int buffer_size
  */
 void delete_fluid_ladspa_fx(fluid_ladspa_fx_t *fx)
 {
-    int i;
+    fluid_list_t *list;
+    fluid_ladspa_node_t *node;
+
     fluid_return_if_fail(fx != NULL);
 
     clear_ladspa(fx);
 
-    /* clear the remaining input or output nodes */
-    for(i = 0; i < fx->num_nodes; i++)
+    /* Delete the remaining host nodes */
+    for(list = fx->host_nodes; list; list = fluid_list_next(list))
     {
-        delete_fluid_ladspa_node(fx->nodes[i]);
+        node = (fluid_ladspa_node_t *) fluid_list_get(list);
+        delete_fluid_ladspa_node(node);
     }
```

Comment:
I think you're missing a call to `delete_fluid_list(fx->host_nodes);` here.

Reply:
Good catch, thanks! Fixed with latest commit.